README.md

```diff
@@ -1,2 +1,120 @@
-# mailgun-relayery
-A Go server providing an additional security layer for using the MailGun API
+Mailgun Relayery
+================
+
+Mailgun Relayery is a tool providing an additional security layer for the MailGun API.
+
+**The problem**: MailGun gives out a single API key to each account. For entities dealing with multiple clients or 
+applications, giving out their API key to each of them constitutes a blatant security violation.
+
+**The solution**: _Mailgun Relayery_ acts as a middle layer between message senders and the MailGun API.
+Senders are issued descriptor strings (acting like usernames) and HTTP authentication tokens, which 
+are used to authenticate their requests to the middle layer and to relay their messages to MailGun.
+
+The tool stores authentication data (HTTP tokens) and channeling data (mailing fields like recipients, cc, bcc, etc) 
+in an LMDB, guaranteeing persistency. Two Go servers have access to the LMDB:
+
+* the Control server (with read-write access to the LMDB) manages authentication data and mailing channels; this 
+    server should only be accessed via secure connection, e.g. ssh, and managed by a trusted party.
+* the Relay server (with read-only access to the LMDB) receives and authenticates HTTP requests to relay 
+    messages to the MailGun API; this server is open to the whole internet.
+
+
+All communication with the servers takes place via HTTP following the
+[relay server API](https://github.com/Parquery/mailgun-relayery/swagger/relay/swagger.yaml) and the
+[control server API](https://github.com/Parquery/mailgun-relayery/swagger/control/swagger.yaml).
+
+An example [Python script](https://github.com/Parquery/mailgun-relayery/example.py) is available for a quick start. 
+
+
+Usage
+=====
+
+Preliminary steps
+-----------------
+*  Run the [release script](https://github.com/Parquery/mailgun-relayery/release.py) to compile and release the two Go 
```

Comment:
s/Preliminary steps/Compilation

Please also note down a TODO so that we don't forget to add a link to the pre-compiled binaries released on github once we publish it.